intelmq/bin/intelmqctl.py

```diff
@@ -1448,32 +1441,21 @@ def check(self, no_connections=False):
-                check_logger.warning('Misconfiguration: No pipeline configuration found for %r.', bot_id)
-            elif bot_id in files[PIPELINE_CONF_FILE]:
-                if ('group' in bot_config and
-                        bot_config['group'] in ['Collector', 'Parser', 'Expert']):
-                    if ('destination-queues' not in files[PIPELINE_CONF_FILE][bot_id] or
-                            (isinstance(files[PIPELINE_CONF_FILE][bot_id]['destination-queues'], list) and
-                             len(files[PIPELINE_CONF_FILE][bot_id]['destination-queues']) < 1) or
-                            (isinstance(files[PIPELINE_CONF_FILE][bot_id]['destination-queues'], dict) and
-                             '_default' not in files[PIPELINE_CONF_FILE][bot_id]['destination-queues'])):
-                        check_logger.error('Misconfiguration: No (default) destination queue for %r.', bot_id)
-                        retval = 1
-                    else:
-                        all_queues = all_queues.union(files[PIPELINE_CONF_FILE][bot_id]['destination-queues'])
-                if ('group' in bot_config and
-                        bot_config['group'] in ['Parser', 'Expert', 'Output']):
-                    if ('source-queue' not in files[PIPELINE_CONF_FILE][bot_id] or
-                            not isinstance(files[PIPELINE_CONF_FILE][bot_id]['source-queue'], str)):
-                        check_logger.error('Misconfiguration: No source queue for %r.', bot_id)
-                        retval = 1
-                    else:
-                        all_queues.add(files[PIPELINE_CONF_FILE][bot_id]['source-queue'])
-                        all_queues.add(files[PIPELINE_CONF_FILE][bot_id]['source-queue'] + '-internal')
+            if ('group' in bot_config and bot_config['group'] in ['Collector', 'Parser', 'Expert']):
+                if ('destination-queues' not in bot_config['parameters'] or
+                   (isinstance(bot_config['parameters']['destination_queues'], list) and len(bot_config['parameters']['destination-queues']) < 1) or
+                   (isinstance(bot_config['parameters']['destination_queues'], dict) and '_default' not in bot_config['parameters']['destination_queues'])):
+                    check_logger.error('Misconfiguration: No (default) destination queue for %r.', bot_id)
+                    retval = 1
+                else:
+                    all_queues = all_queues.union(bot_config['parameters']['destination-queues'])
+            if ('group' in bot_config and bot_config['group'] in ['Parser', 'Expert', 'Output']):
+                if ('source-queue' in bot_config['parameters'] and instance(bot_config['parameters']['source_queue'], str)):
+                    all_queues.add(bot_config['parameters']['source_queue'])
+                    all_queues.add(f"{bot_config['parameters']['source_queues']}-internal")
+                else:
+                    all_queues.add(f"{bot_id}-source")
```

Comment:
In bot.py the queues are called `$botid-queue`

Reply:
Fixed